main.go

```diff
@@ -639,19 +751,13 @@ func POST_Handling(c *gin.Context) {
 
 		msg.DisableWebPagePreview = true
 
-		sendmsg, err := bot.Send(msg)
-		if err == nil {
-			c.String(http.StatusOK, "telegram msg sent.")
-		} else {
-			log.Printf("Error sending message: %s", err)
-			c.JSON(http.StatusServiceUnavailable, gin.H{
-				"err":     fmt.Sprint(err),
-				"message": sendmsg,
-				"srcmsg":  fmt.Sprint(msgtext),
-			})
```

Comment:
Why did you remove this code? I can see similar one in `HandleIncomingAlerts` function but it doesn't return error message from `bot.Send`.

Reply:
@hryamzik I moved the actual handling and Telegram message sending to `HandleIncomingAlerts`, so that it was available for the deadmanSwitchChecker goroutine as well. However, that goroutine does not run inside a HTTP context, so I had to take the HTTP responding outside of it and inside `POST_handling`, where I think it also conceptually belongs.

From `HandleIncomingAlerts` now idiomatically, I would have to return an `err` to indicate the Telegram message failed to send. This means I would typically wrap any failure information inside the `err`, which I assumed `bot.Send()` already does for me. I realize this splitup loses the feature of the `srcmsg` and `message` in the JSON error response; I leave it up to you whether that is acceptable. I can either leave it like this, add those two to the returned `err`, or return them separately.

Reply:
You can add an extra `*gin.Context` argument to `HandleIncomingAlerts` and keep original login when it's not `nil`.